src/internal_events/http_client.rs

```diff
@@ -0,0 +1,91 @@
+use super::InternalEvent;
+use http::{Request, Response};
+use hyper::{body::HttpBody, Error};
+use metrics::{counter, histogram};
+use std::time::Duration;
+
+#[derive(Debug)]
+pub struct AboutToSendHTTPRequest<'a, T> {
+    pub request: &'a Request<T>,
+}
+
+impl<'a, T: HttpBody> InternalEvent for AboutToSendHTTPRequest<'a, T> {
+    fn emit_logs(&self) {
+        debug!(
+            message = "Sending HTTP request.",
+            uri = %self.request.uri(),
+            method = %self.request.method(),
+            version = ?self.request.version(),
+            headers = ?self.request.headers(),
+            body = %FormatBody(self.request.body()),
+        );
+    }
+
+    fn emit_metrics(&self) {
+        counter!("http_client_requests_sent_total", 1, "method" => self.request.method().to_string());
```

Comment:
```suggestion
        counter!("http_client_requests_sent_total", 1, "method" => self.request.method().as_str());
```
As I understand it's not required crate `String` for counter and [as_str](https://docs.rs/hyper/0.13.9/hyper/struct.Method.html#method.as_str) can be used.

Reply:
Nope, the value has to be `Cow<'static, str>` under the layers of wrappers, so this is effectively the shortest path to build that.